lib/config.js

```diff
@@ -45,6 +45,17 @@ export default function subscribeConfig(conn, configChanged) {
       updateConfig({ services: Object.assign({}, config.services, { [domain]: domainInfo }) });
     }
 
+    function processServiceRemoved(event) {
+      if (config === null) return;
+
+      const { domain, service } = event.data;
+
+      if (!config.services[domain] || !config.services[domain][service]) return;
+
+      delete config.services[domain][service];
```

Comment:
- We should never mutate the existing object but always create a new one
 - Don't use `delete`. [It causes de-optimizations in V8](https://www.smashingmagazine.com/2012/11/writing-fast-memory-efficient-javascript/#de-referencing-misconceptions)